src/inference/src/dev/icompiled_model.cpp

```diff
@@ -143,3 +143,7 @@ ov::SoPtr<ov::IRemoteContext> ov::ICompiledModel::get_context() const {
         return m_context;
     return m_plugin->get_default_context({});
 }
+
+void ov::ICompiledModel::set_model_shared_object(ov::Model& model, const std::shared_ptr<void>& shared_object) {
+    model.m_shared_object = shared_object;
```

Comment:
Hi, my consideration is that `Model::m_shared_object` is a private member. If we do not provide a function or method to set the `m_shared_object`, we have to make the `CompiledModel` classes in each plugin a friend of the `ov::Model` class, since friendship cannot be inherited.

Reply:
good point